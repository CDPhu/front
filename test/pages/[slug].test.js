import renderer from 'react-test-renderer';
import Post, { getStaticProps, getStaticPaths } from '../../pages/aktuelles/[slug]';

jest.mock('next/image');
jest.mock('../../components/navbar');
jest.mock('../../components/head');
jest.mock('html-react-parser');
jest.mock('../../components/newsletter');
jest.mock('../../components/footer');
jest.mock('@heroicons/react/24/solid');
jest.mock('../../lib/directus');

const renderTree = tree => renderer.create(tree);
describe('<Post>', () => {
  it('should render component', () => {
    expect(renderTree(<Post />).toJSON()).toMatchSnapshot();
  });
  
});


describe('getStaticProps', () => {
  it('should expose a function', () => {
		expect(getStaticProps).toBeDefined();
	});
  
  it('getStaticProps should return expected output', async () => {
    // const retValue = await getStaticProps();
    expect(true).toBeTruthy;
  });
});

describe('getStaticPaths', () => {
  it('should expose a function', () => {
		expect(getStaticPaths).toBeDefined();
	});
  
  it('getStaticPaths should return expected output', async () => {
    const retValue = await getStaticPaths();
    expect(retValue).toBeTruthy;
  });
});