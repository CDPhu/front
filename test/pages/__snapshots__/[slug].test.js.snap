// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post> should render component 1`] = `
<div
  className="pt-12 min-h-screen overflow-hidden"
>
  <div
    className="relative max-w-7xl mx-auto py-16 px-4 sm:px-6 lg:px-8"
  >
    <div
      className="mx-auto text-base max-w-prose lg:grid lg:grid-cols-2 lg:gap-8 lg:max-w-none"
    >
      <div>
        <h2
          className="text-bvpk-800 font-titillium tracking-wide font-bold uppercase"
        />
        <h3
          className="mt-2 text-3xl leading-8 font-titillium text-gray-700 font-bold sm:text-4xl"
        />
      </div>
    </div>
    <div
      className="mt-8 lg:grid lg:grid-cols-2 lg:gap-8"
    >
      <div
        className="relative lg:row-start-1 lg:col-start-2"
      >
        <div
          className="relative text-base mx-auto max-w-prose lg:max-w-none"
        >
          <figure>
            <div
              className="relative aspect-w-12 aspect-h-7 lg:aspect-none"
            />
            <figcaption
              className="mt-3 flex text-sm text-gray-500"
            >
              <span
                className="ml-2"
              >
                Bild von
                 
                <a
                  className=""
                >
                  <a
                    href="/impressum"
                  />
                </a>
              </span>
            </figcaption>
          </figure>
        </div>
      </div>
      <div
        className="mt-8 lg:mt-0"
      >
        <div
          className="prose prose-lg prose-on-bvpk-aktuelles max-w-prose mx-auto lg:max-w-none"
        >
          There has been an error. Sorry about that
        </div>
      </div>
    </div>
  </div>
</div>
`;
